src/server/inline.preboot.code.ts

```diff
@@ -28,19 +28,19 @@ export const defaultOptions = <PrebootRecordOptions>{
       freeze: true
     },
 
+    // when user submit form (press enter, click on button/input[type="submit"])
+    {
+      selector: 'form',
+      events: ['submit'],
+      preventDefault: true,
+      freeze: true
+    },
+
     // for tracking focus (no need to replay)
     {
       selector: 'input,textarea',
       events: ['focusin', 'focusout', 'mousedown', 'mouseup'],
       noReplay: true
-    },
-
-    // user clicks on a button
-    {
```

Comment:
Sorry, one more change. for the same reason as the other change we still need in an event listener for buttons. I think your form submission event listener will cover the `input[type="submit"]` but not a button which has no form. So, just add back in:

```
{
  selector: 'button',
  events: ['click'],
  preventDefault: true,
  freeze: true
}
```